File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    zero_count = lottos.count(0)
    
    match_count = len(set(lottos) & set(win_nums))
    
    rank = [6, 6, 5, 4, 3, 2, 1]
    
    return [rank[match_count + zero_count], rank[match_count]]

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <unordered_set>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    int zero_count = count(lottos.begin(), lottos.end(), 0);
    
    unordered_set<int> w(win_nums.begin(), win_nums.end());
    
    int max_count = 0;
    for (int num : lottos) {
        if (num != 0 && w.find(num) != w.end()) {
            max_count++;
        }
    }
    
    vector<int> rank = {6, 6, 5, 4, 3, 2, 1};
    
    return {rank[max_count + zero_count], rank[max_count]};
}